x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/dev_console.tsx

```diff
@@ -15,82 +23,150 @@ import {
   Console,
   RegisteredConsoleClient,
   useConsoleManager,
 } from '../../../components/console';
 
 const delay = async (ms: number = 4000) => new Promise((r) => setTimeout(r, ms));
 
-class DevCommandService implements CommandServiceInterface {
-  getCommandList(): CommandDefinition[] {
-    return [
-      {
-        name: 'cmd1',
-        about: 'Runs cmd1',
-      },
-      {
-        name: 'get-file',
-        about: 'retrieve a file from the endpoint',
-        args: {
-          file: {
-            required: true,
-            allowMultiples: false,
-            about: 'the file path for the file to be retrieved',
-          },
-        },
+const getCommandList = (): CommandDefinition[] => {
+  return [
+    {
+      name: 'cmd1',
+      about: 'Runs cmd1',
+      RenderComponent: ({ command, setStatus, store, setStore }) => {
+        const isMounted = useIsMounted();
+
+        const [apiResponse, setApiResponse] = useState<null | string>(null);
+        const [uiResponse, setUiResponse] = useState<null | ReactElement>(null);
+
+        // Emulate a real action where:
+        // 1. an api request is done to create the action
+        // 2. wait for a response
+        // 3. account for component mount/unmount and prevent duplicate api calls
+
+        useEffect(() => {
+          (async () => {
+            // Emulate an api call
+            if (!store.apiInflight) {
+              setStore({
+                ...store,
+                apiInflight: true,
+              });
+
+              window.console.warn(`${Math.random()} ------> cmd1: doing async work`);
+
+              await delay(6000);
+              setApiResponse(`API was called at: ${new Date().toLocaleString()}`);
+            }
+          })();
+        }, [setStore, store]);
+
+        useEffect(() => {
+          (async () => {
+            const doUiResponse = () => {
+              setUiResponse(
+                <EuiText>
+                  <EuiText>{`${command.commandDefinition.name}`}</EuiText>
+                  <EuiText>{`command input: ${command.input}`}</EuiText>
+                  <EuiText>{'Arguments provided:'}</EuiText>
+                  <EuiCode>{JSON.stringify(command.args, null, 2)}</EuiCode>
+                </EuiText>
+              );
+            };
+
+            if (store.apiResponse) {
+              doUiResponse();
+            } else {
+              await delay();
+              doUiResponse();
+            }
+          })();
+        }, [
+          command.args,
+          command.commandDefinition.name,
+          command.input,
+          isMounted,
+          store.apiResponse,
+        ]);
+
+        useEffect(() => {
+          if (apiResponse && uiResponse) {
+            setStatus('success');
+          }
+        }, [apiResponse, setStatus, uiResponse]);
+
+        useEffect(() => {
+          if (apiResponse && store.apiResponse !== apiResponse) {
+            setStore({
+              ...store,
+              apiResponse,
```

Comment:
I presume we will abtract this as we move forward? Else this is kind of hard to read.

Reply:
this entire file will be deleted soon. it was here only while I developed the console since we don't currently have an entry point into it. When the `isolate` action PR merges, I should be able to delete this.